hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/DataSourceOptions.scala

```diff
@@ -138,6 +138,14 @@ object DataSourceReadOptions {
         " read from the data file (in Hudi partition columns are persisted by default)." +
         " This config is a fallback allowing to preserve existing behavior, and should not be used otherwise.")
 
+  val REFRESH_PARTITION_AND_FILES_IN_INITIALIZATION: ConfigProperty[Boolean] =
```

Comment:
I don't think we need a config for that, instead:

- Lazy loading should be default behavior
- We should add a config that can override default behavior and always load eagerly (fallback)

Reply:
Do you mean setting the config to false? Indeed, I intend to have this option as a fallback.

Reply:
@YuweiXiao the other way around:
 - We should list lazily by default
 - We should add config to override this behavior and fallback to eager listing (if necessary)

Reply:
OK, will turn it on by default.